router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/fetchJobStatus.go

```diff
@@ -0,0 +1,205 @@
+		"Content-Type":  JSONMimeType,
+		"Authorization": fmt.Sprintf("Bearer %s", accessToken),
+	}
+
+	if jobType == "fail" {
+		url = fmt.Sprintf("https://%s.mktorest.com/bulk/v1/leads/batch/%s/failures.json", munchkinId, importId)
+	} else {
+		url = fmt.Sprintf("https://%s.mktorest.com/bulk/v1/leads/batch/%s/warnings.json", munchkinId, importId)
+	}
+
+	startTime := time.Now()
+	resp, err := handleHttpRequest("GET", url, headers)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	endTime := time.Now()
+	requestTime := endTime.Sub(startTime).Milliseconds()
+
+	fmt.Printf("Request time: %d ms\n", requestTime)
+
+	if !isHttpStatusSuccess(resp.StatusCode) {
+		return "", fmt.Errorf("could not fetch job status: received HTTP status %d", resp.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func handleFetchJobStatusResponse(resp string, jobType string) (string, error) {
+	// Handle the response as needed, this function is a placeholder
+	return resp, nil
+}
+
+func responseHandler(event MarketoAsyncFailedPayload, jobType string) (*Response, error) {
+	config := event.Config
+	accessToken, err := getAccessToken(config)
+	if err != nil {
+		return nil, err
+	}
+
+	jobStatus, err := getJobsStatus(event, jobType, accessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	jobStatusArr := strings.Split(jobStatus, "\n")
+	headerArr := strings.Split(event.MetaData.CSVHeaders, ",")
+
+	if len(headerArr) == 0 {
+		return nil, &PlatformError{"No csvHeader in metadata"}
+	}
+
+	data := make(map[int64]string)
+
+	fieldSchemaMapping, err := getFieldSchemaMap(accessToken, config["MunchkinId"])
+	if err != nil {
+		return nil, err
+	}
+
+	unsuccessfulJobInfo := checkEventStatusViaSchemaMatching(event, fieldSchemaMapping)
+	mismatchJobIdArray := make([]int64, 0, len(unsuccessfulJobInfo))
+	for k := range unsuccessfulJobInfo {
+		if jobID, err := strconv.ParseInt(k, 10, 64); err == nil {
+			mismatchJobIdArray = append(mismatchJobIdArray, jobID)
+		}
+	}
+
+	reasons := unsuccessfulJobInfo
+
+	filteredInputs := make([]MarketoAsyncFailedInput, 0, len(event.Input))
+	for _, input := range event.Input {
+		if !contains(mismatchJobIdArray, input.Metadata.JobID) {
+			filteredInputs = append(filteredInputs, input)
+		}
+	}
+
+	for _, input := range filteredInputs {
+		responses := make([]string, 0, len(headerArr))
+		for _, fieldName := range headerArr {
+			if val, ok := input.Message[fieldName]; ok {
+				responses = append(responses, fmt.Sprintf("%v", val))
+			}
+		}
+		data[input.Metadata.JobID] = strings.Join(responses, ",")
+	}
+
+	unsuccessfulJobIdsArr := make([]int64, 0)
+	successfulJobIdsArr := make([]int64, 0)
+
+	for _, element := range jobStatusArr {
+		elemArr := parseCSVLine(element)
+		reasonMessage := elemArr[len(elemArr)-1]
+		elemArr = elemArr[:len(elemArr)-1]
+
+		for key, val := range data {
+			if val == strings.Join(elemArr, ",") {
```

Comment:
we should use the lookup field for comparison not compare the whole record
https://developer.adobe.com/marketo-apis/api/mapi/#tag/Bulk-Import-Leads

Reply:
There can be one possible change in the functionality. The user needs to ensure that lookup field should be chosen as primary key while configuring the source.
Here is the reason:
Let's say lookup field is Email, and while configuring the source if we choose ID as a primary key, then source can query the rows where id must be present but email can be null / empty string etc. 
In that case,  Marketo throws some warning. 
I was sending the below data in csv:
```
FirstName,LastName,Email,Company
'Able','Baker','', 'marketo'
Able,"Baker","shrouti@gmail.com", "marketo"
```
And while polling I got response :
```
{
    "requestId": "14cf8#1912627bf66",
    "result": [
        {
            "batchId": 7271,
            "importId": "7271",
            "status": "Complete",
            "numOfLeadsProcessed": 2,
            "numOfRowsFailed": 0,
            "numOfRowsWithWarning": 1,
            "message": "Import succeeded, 2 records imported (2 members), 1 warning."
        }
    ],
    "success": true
}
```
For this the code flow will appear in fetch job status part, and if we search with only lookup field, we will not get anything to refer from. Previously we were marking these kind of warnings as failure, but here, it will be automatically marked as success.

The response : 

```
FirstName,LastName,Email,Company,Import Warning Reason
'Able','Baker','', 'marketo',Invalid email address
```
@koladilip let me know if you need some more clarification.

Reply:
this is a rare case and it is fine, we can update a footer note in the form, I don't think this breaks our logic so we are good